pkg/operator/loglevel/logging_controller.go

```diff
@@ -28,24 +35,37 @@ func (c LogLevelController) sync(ctx context.Context, syncCtx factory.SyncContex
 		return err
 	}
 
-	currentLogLevel := CurrentLogLevel()
+	currentLogLevel, isUnknown := c.currentLogLevelFn()
 	desiredLogLevel := detailedSpec.OperatorLogLevel
 
+	// if operator operatorSpec OperatorLogLevel is empty, default to "Normal"
+	// TODO: This should be probably done by defaulting the CR field?
 	if len(desiredLogLevel) == 0 {
 		desiredLogLevel = operatorv1.Normal
 	}
 
-	// When the current loglevel is the desired one, do nothing
-	if currentLogLevel == desiredLogLevel {
+	// correct log level is set and it matches the expected log level from operator operatorSpec, do nothing.
+	if isUnknown && currentLogLevel == desiredLogLevel {
 		return nil
 	}
 
-	// Set the new loglevel if the operator spec changed
-	if err := SetVerbosityValue(desiredLogLevel); err != nil {
+	// log level is not specified in operatorSpec and the log verbosity is not set (0), default the log level to V(2).
+	if len(desiredLogLevel) == 0 {
+		desiredLogLevel = currentLogLevel
+	}
+
+	// Set the new loglevel if the operator operatorSpec changed
+	if err := c.setVerbosityFn(desiredLogLevel); err != nil {
 		syncCtx.Recorder().Warningf("OperatorLoglevelChangeFailed", "Unable to change operator log level from %q to %q: %v", currentLogLevel, desiredLogLevel, err)
 		return err
 	}
 
+	// give better message when we setting the log level for the first time
+	if !isUnknown {
+		syncCtx.Recorder().Eventf("OperatorLoglevelChange", "Operator log level set to %q", currentLogLevel)
```

Comment:
How do we make sure this does not fire on every restart?

Reply:
it will. we can skip this event.

Reply:
yes, please. If you can contain it, fine. But it must not show up multiple times, only on CRD user action.